apps/api/src/controllers/admin.controller.ts

```diff
@@ -797,16 +797,21 @@ export function makeAdminController(services: ServiceRegistry) {
           );
         }
 
-        // Extract rewards and tradingConstraints from the validated data
-        const { rewards, tradingConstraints, ...competitionUpdates } =
-          bodyResult.data;
+        // Extract rewards, tradingConstraints, and perpsProvider from the validated data
+        const {
+          rewards,
+          tradingConstraints,
+          perpsProvider,
+          ...competitionUpdates
+        } = bodyResult.data;
         const updates = competitionUpdates as UpdateCompetition;
```

Comment:
I think we can remove this `as` in favor of parsing with Zod if we update the `AdminCreateCompetitionSchema` to validate and coerce iso data strings into dates, which will work with the Drizzle UpdateCompetition type.

Specifically, use `z.iso.datetime().pipe(z.coerce.date()).optional()`

Reply:
Nice, clean addition - updated in [9705e13](https://github.com/recallnet/js-recall/pull/1274/commits/9705e1365d5a49357d61d68747d9f2979bcdd5e7)